cloud-init expects /proc/1/environ to be =-separated key-value pairs, but in
our CI runners it's actually `-D -e [listener] 0 of 10-100 startups` because
our PID 1 is sshd: we now just skip over this invalid data.

diff --git a/cloudinit/util.py b/cloudinit/util.py
index 3e65c7e..310f8a8 100644
--- a/cloudinit/util.py
+++ b/cloudinit/util.py
@@ -2481,7 +2484,7 @@ def get_proc_env(
     contents = contents.decode(encoding, errors)
 
     for tok in contents.split(null):
-        if not tok:
+        if not tok or equal not in tok:
             continue
         (name, val) = tok.split(equal, 1)
         if name:
